ui/app/components/job-editor.js

```diff
@@ -12,12 +12,39 @@ export default class JobEditor extends Component {
 
   @tracked error = null;
   @tracked planOutput = null;
+  @tracked view = 'job-spec';
+  @tracked isEditing;
+
+  constructor() {
+    super(...arguments);
+    this.isEditing = !!(this.args.context === 'new');
+  }
+
+  toggleEdit(bool) {
+    this.isEditing = bool || !this.isEditing;
+  }
+
+  @action
+  edit() {
+    this.args.job.set(
+      '_newDefinition',
+      JSON.stringify(this.args.definition, null, 2)
+    );
+    this.toggleEdit(true);
+  }
+
+  @action
+  onCancel() {
+    this.toggleEdit(false);
+  }
 
   get stage() {
-    return this.planOutput ? 'plan' : 'editor';
+    if (this.planOutput) return 'review';
+    if (this.isEditing) return 'edit';
+    else return 'read';
   }
```

Comment:
Request: Now that this is becoming a more complicated stateful page, I'd like to see that state represented as queryParams in the URL somehow.

`jobs/:jobid/definition?mode=edit&spec=hcl` vs `jobs/:jobid/definition?mode=read&spec=json` and the like.

Reply:
This is a component, not a controller. I don't think URL based state makes sense here because this component is rendered across multiple views and meant to expand.

Reply:
Well why not have `stage` exist as a property that gets passed down to this component? `<JobEditor @stage={{this.qp.stage}} ...` etc. from the controller?

and you'd want to transform your `get stage()` in this component to something like

```
  get stage() {
    return this.args.stage || <your current logic>
  }
```

Reply:
I could totally do that, this PR is for separating the concerns of state and presentation logic. I could write a separate PR for updating the component's API and changing this to be addressable state flowing in from the controller, instead of UI state on the component would require a refactoring of the associated routes and controllers everywhere this component is invoked. So it will require its own PR because that's a larger change that shouldn't scope creep into this PR.